test/com/google/javascript/jscomp/Es6VariableReferenceCheckTest.java

```diff
@@ -327,12 +327,15 @@ public void testObjectPattern_defaultValue() {
    * basic checks.
    */
   public void testDefaultParam() {
-    assertEarlyReferenceError("function f(x=a) {}");
     assertEarlyReferenceError("function f(x=a) { let a; }");
+    assertEarlyReferenceError(LINE_JOINER.join(
+        "function f(x=a) { let a; }",
+        "function g(x=1) { var a; }"));
     assertEarlyReferenceError("function f(x=a) { var a; }");
     assertEarlyReferenceError("function f(x=a()) { function a() {} }");
     assertEarlyReferenceError("function f(x=[a]) { var a; }");
     assertEarlyReferenceError("function f(x=y, y=2) {}");
+    assertNoWarning("function f(x=a) {}");
```

Comment:
The VarCheck pass would give us a warning here right? Just not the Es6VariableReferenceCheck pass?